app/lib/analyzer/backend.dart

```diff
@@ -92,4 +99,47 @@ class AnalysisBackend {
       await tx.commit();
     });
   }
+
+  /// Whether [packageName] with [packageVersion] exists, and
+  /// - it has no recent [Analysis] with the given [analysisVersion], or
+  /// - it has no newer [Analysis] than [analysisVersion].
+  Future<bool> isValidTarget(
+      String packageName, String packageVersion, String analysisVersion) async {
+    if (packageName == null || packageVersion == null) return false;
+    final List<PackageVersion> versions = await db.lookup([
+      db.emptyKey
+          .append(Package, id: packageName)
+          .append(PackageVersion, id: packageVersion)
+    ]);
+    // package or version does not exists
+    if (versions.isEmpty || versions.first == null) return false;
+
+    final List<PackageAnalysis> list =
+        await db.lookup([db.emptyKey.append(PackageAnalysis, id: packageName)]);
+    // no analysis for the package
+    if (list.isEmpty || list.first == null) return true;
+
+    final PackageAnalysis packageAnalysis = list.first;
+    final PackageVersionAnalysis versionAnalysis =
+        (await db.lookup([packageAnalysis.latestVersionKey])).first;
+
+    if (versionAnalysis.analysisVersion == analysisVersion) {
+      final DateTime now = new DateTime.now();
```

Comment:
To make the reader not care about whether `DateTime.difference` automatically converts one argument to utc if the other is utc already, you could make `now` be utc.